### PR TITLE
Per repo hook config

### PR DESCRIPTION
**Pull request by lrvick**

### CODE REVIEW

File: bin/mkrepo [hunk 1]
```diff
@@ -30,23 +30,39 @@ cat <<- "EOF" > .git/hooks/pre-receive
...
+			hook_dir="/git/hooks"
+			if [ ! -d "$hook_dir" ]; then
+				mkdir -p $hook_dir
+				git clone $HOOK_REPO $hook_dir
+			fi
+			if [ ! -z "$HOOK_REPO_REF" ]; then
+				git \
+					--git-dir="$hook_dir/.git" \
+					--work-tree="$hook_dir" \
+					checkout $HOOK_REPO_REF
```
Comment (thepwagner) [inline bin/mkrepo]: Checkout before fetch is confusing order to me. if HOOK_REPO_REF is a branch i just added, the checkout will fail?

**Comment by thepwagner** [comment]
Are tests stable enough for travis-ci?

**Comment by lrvick** [comment]
The tests are stable, however I doubt Travis will handle docker correctly. Lot of reported issues with that.

**Comment by lrvick** [comment]
Also reviewers: @thepwagner and/or @AvianFlu

**Comment by thepwagner** [comment]
+1, only concerns are skin deep

**Comment by thepwagner** [comment]
Tests verified downstream.

**Pull request merged**